### ISSUE TITLE
x86 debugging does not work if the debugger binary view is written in the correct way

### ISSUE THREAD

**Issue by xusheng6** [Vector35/debugger#392]
In 7a8c92ac2b519f3a26a30244906c4b66b12e0b92, I fixed the implementation of debugger binary view, and it fixes issue #334. However, I soon realize the correct implementation of the debugger binary view causes several issues:

1. https://github.com/Vector35/debugger/issues/386
2. https://github.com/Vector35/debugger/issues/385
3. https://github.com/Vector35/debugger/issues/384

These issues happen because they expose certain erroneous handling of the huge memory space.

To address these issues, I have to temporarily revert the commit as done in 7a8c92ac2b519f3a26a30244906c4b66b12e0b92. 

To fix the above issues, we will need #96, which is not trivial to do.

**Comment by xusheng6** [comment]
After reverting the fix, we will get error messages like 
```
Truncating segment @ 0x0 with dataOffset @ 0x0 and length 0xffffffffffffffff to 0x4058 bytes.
```
in the console

**Comment by xusheng6** [comment]
This is fixed in 6d4820960db1db220eac788f18e535d9617370eb
